### PR TITLE
🐛 Add reference hash and others to metadata arguments

### PR DESCRIPTION
**Pull request by microchipgnu**

### CODE REVIEW

**Review by sainthiago — APPROVED** [review]

**Review by tifrel — CHANGES_REQUESTED** [review]

File: src/wallet.ts [hunk 1]
```diff
@@ -1087,9 +1089,13 @@ export class Wallet {
...
       owner_id: accountId,
       metadata: {
+        title: metadata?.title,
+        description: metadata?.description,
+        media: metadata?.media,
+        media_hash: metadata?.media_hash,
         reference: metadataId,
-        // TODO: check if category is lowercase
-        extra: !category ? null : category,
+        reference_hash: metadataId,
```
[the remark this answers is not in the document]
Comment (microchipgnu) [reply src/wallet.ts:1097]: Care to clarify?

**Review by cif — APPROVED** [review]

File: src/utils/crypto.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const hash = async (buffer: ArrayBuffer) => {
+  const hashBuffer = await crypto.subtle.digest('SHA-256', buffer)
+  const hashArray = Array.from(new Uint8Array(hashBuffer))
+  const base64 = btoa(
+    hashArray.map((b) => b.toString(16).padStart(2, '0')).join('')
```
Comment (tifrel) [inline src/utils/crypto.ts:5]: Interesting, why the base16 in the middle?

File: src/wallet.ts [hunk 3]
```diff
@@ -1087,9 +1090,13 @@ export class Wallet {
     const obj = {
       owner_id: accountId,
       metadata: {
+        title: metadata?.title,
+        description: metadata?.description,
+        media: metadata?.media,
+        media_hash: metadata?.media_hash,
```
Comment (microchipgnu) [inline src/wallet.ts:1096]: Shall we keep these here?

@tifrel @cif

**Review by tifrel — APPROVED** [review]

File: src/wallet.ts (hunk 3 above)
Comment (tifrel) [reply src/wallet.ts:1096]: Why not. But we need to be sure that if media is set, media_hash is set as well

Comment (microchipgnu) [reply src/wallet.ts:1096]: Yes, when uploading through MintbaseJS `media_hash` is always generated.

**Comment by cif** [comment]
Closing this, we'll revisit as we make refactoring changes to the minter

**Pull request closed without merging**